corehq/apps/app_manager/util.py

```diff
@@ -607,3 +608,42 @@ def use_app_aware_sync(app):
     Determines whether OTA restore should sync only cases/ledgers/fixtures of the given app where possible
     """
     return toggles.APP_AWARE_SYNC.enabled(app.domain)
+
+
+def _prepare_xpath_for_validation(xpath):
+    prepared_xpath = xpath.lower()
+    prepared_xpath = prepared_xpath.replace('"', "'")
+    prepared_xpath = re.compile('\s').sub('', prepared_xpath)
+    return prepared_xpath
+
+
+def xpath_references_case(xpath):
+    if not isinstance(xpath, basestring):
+        return False
+
+    prepared_xpath = _prepare_xpath_for_validation(xpath)
+
+    return any([
+        case_reference in prepared_xpath for case_reference in [
+            "instance('casedb')",
+            "session/data/case_id",
+        ]
+    ])
+
+
+def form_filter_references_case(xpath):
+    if not isinstance(xpath, basestring):
+        return False
+
+    prepared_xpath = _prepare_xpath_for_validation(xpath)
+
+    return any([
+        case_reference in prepared_xpath for case_reference in [
+            "#case",
+            "#parent",
+            "#host",
```

Comment:
(or is the user case not relevant here because it'll be in the session and not the case DB?)